keanu-project/src/test/java/io/improbable/keanu/util/json/JsonTest.java

```diff
@@ -0,0 +1,74 @@
+package io.improbable.keanu.util.json;
+
+import com.google.protobuf.util.JsonFormat;
+import io.improbable.keanu.KeanuSavedBayesNet;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.closeTo;
+import static org.hamcrest.collection.IsIn.isIn;
+import static org.hamcrest.core.Every.everyItem;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+public class JsonTest {
+
+    private static BayesianNetwork net;
+    private static Map<String, String> someMetadata;
+    private static ByteArrayOutputStream output;
+
+    @BeforeClass
+    public static void setup() throws IOException {
+        DoubleVertex gaussianVertex = new GaussianVertex(0.0, 1.0);
+        gaussianVertex.observe(0.5);
+        net = new BayesianNetwork(gaussianVertex.getConnectedGraph());
+
+        output = new ByteArrayOutputStream();
+
+        someMetadata = new HashMap<>();
+        someMetadata.put("Author", "Some Author");
+        someMetadata.put("Tag", "MyBayesNet");
+
+        JsonSaver jasonSaver = new JsonSaver(net, someMetadata);
+        jasonSaver.save(output, true);
+    }
+
+    @Test
+    public void metadataIsSavedForJsonModels() throws IOException {
```

Comment:
We are testing that `JsonSaver` saved the metadata to `output` right? Can we rename the test to something like "JsonSaverSavesMetadata"

Also the word `JsonModel` confused me for a bit.

Reply:
Also might be useful to see a conversion of the output stream to a Json object and/or Json string and compare that to an expected.

Reply:
Yup, `jsonSaverSavesMetadata` makes more sense. Thanks! :)

I'll create a new test for comparison with expected json output.

Reply:
Done, let me know what you think!